src/main/java/com/dao/UserDAO.java

```diff
@@ -0,0 +1,205 @@
+package com.dao;
+
+import com.model.User;
+
+import javax.sql.DataSource;
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+public class UserDAO {
+
+    private static UserDAO instance;
+    private DataSource dataSource;
+
+    private static final String IS_EXIST_QUERY = "SELECT * FROM users WHERE user_name = ?";
+    private static final String VALIDATION_QUERY = "SELECT * FROM users WHERE user_name = ? AND password_hash = ?";
+    private static final String GET_BY_NAME_QUERY = "SELECT * FROM users WHERE user_name = ?";
+    private static final String ADD_QUERY = "INSERT INTO users VALUES (NULL, ?, ?, ?)";
+    private static final String REMOVE_QUERY = "DELETE FROM users WHERE user_name = ?";
+    private static final String UPDATE_QUERY = "UPDATE users SET is_admin = ?, password_hash = ? WHERE user_name = ?";
+    private static final String GET_ALL_QUERY = "SELECT * FROM users";
+
+    private UserDAO() {
+    }
+
+    public static UserDAO getInstance() {
+
+        if (instance == null) {
+
+            instance = new UserDAO();
+        }
+
+        return instance;
+    }
+
+    public void setDataSource(DataSource dataSource) {
+
+        this.dataSource = dataSource;
+    }
+
+    public boolean isExist(String userName) {
+
+        boolean isPresent = false;
```

Comment:
Actually it's unnecessary to store values in this value. You can return true from try{} block and false from the and of the method